encoding/protobuf/v2/outbound_test.go

```diff
@@ -0,0 +1,195 @@
+			require.NoError(t, err)
+			anyMsg.TypeUrl = tt.anyURL // update to custom URL
+
+			gotMessage, err := client.Call(context.Background(), "", anyMsg, newReq)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.True(t, proto.Equal(testMessage, gotMessage)) // we expect the actual type behind the Any
+			}
+		})
+	}
+}
+
+func TestOutboundWithKnownProtoMsg(t *testing.T) {
+	t.Run("known proto message", func(t *testing.T) {
+		newReq := func() proto.Message { return &testpb.TestMessage{} }
+		trans := yarpctest.NewFakeTransport()
+		// outbound that echos the body back
+		out := trans.NewOutbound(nil, yarpctest.OutboundCallOverride(
+			yarpctest.OutboundCallable(func(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+				return &transport.Response{Body: ioutil.NopCloser(req.Body)}, nil
+			}),
+		))
+
+		client := v2.NewClient(v2.ClientParams{
+			ClientConfig: &transport.OutboundConfig{
+				Outbounds: transport.Outbounds{
+					Unary: out,
+				},
+			},
+			Options: []v2.ClientOption{},
+		})
+
+		testMessage := &testpb.TestMessage{Value: "foo-bar-baz"}
+		gotMessage, err := client.Call(context.Background(), "", testMessage, newReq)
+		require.NoError(t, err)
+		assert.True(t, proto.Equal(testMessage, gotMessage))
+
+	})
+}
+
+func TestOutboundWithAnyProtobufMsg(t *testing.T) {
+	t.Run("any message without resolver", func(t *testing.T) {
+		newReq := func() proto.Message { return &anypb.Any{} }
+		trans := yarpctest.NewFakeTransport()
+		// outbound that echos the body back
+		out := trans.NewOutbound(nil, yarpctest.OutboundCallOverride(
+			yarpctest.OutboundCallable(func(ctx context.Context, req *transport.Request) (*transport.Response, error) {
+				return &transport.Response{Body: ioutil.NopCloser(req.Body)}, nil
+			}),
+		))
+
+		client := v2.NewClient(v2.ClientParams{
+			ClientConfig: &transport.OutboundConfig{
+				Outbounds: transport.Outbounds{
+					Unary: out,
+				},
+			},
+			Options: []v2.ClientOption{},
+		})
+
+		testMessage := &testpb.TestMessage{Value: "foo-bar-baz"}
+		anyMsg, err := anypb.New(testMessage)
+		require.NoError(t, err)
+
+		gotMessage, err := client.Call(context.Background(), "", anyMsg, newReq)
+		require.NoError(t, err)
+		returnMsg := &testpb.TestMessage{}
+		anypb.UnmarshalTo(gotMessage.(*anypb.Any), returnMsg, proto.UnmarshalOptions{})
+		assert.True(t, proto.Equal(testMessage, returnMsg))
+
+	})
+}
+
+type testAnyResolver struct {
+	NewMessage proto.Message
+}
+
+func (r testAnyResolver) FindMessageByName(message protoreflect.FullName) (protoreflect.MessageType, error) {
+	return r.FindMessageByURL(string(message))
+}
+
+func (r testAnyResolver) FindMessageByURL(url string) (protoreflect.MessageType, error) {
+	// Custom resolver for TestMessage resolve with both global registered or custom URL
+	if r.NewMessage != nil {
+		if url == "uber.yarpc.encoding.protobuf.TestMessage" || url == "foo.bar.baz" {
```

Comment:
nit: we can take exact url from the test case and store it in testAnyResolver struct